### PR TITLE
Fix/mkt 4222 lint issues

### PR DESCRIPTION
**Pull request by Amitkanswal**
## Changes-
- Snyk fixes
- Removed unused dependencies 
- Fixed TS issues
- Some fixable lint issues
- Resolved warnings on build

### CODE REVIEW

File: .github/workflows/jira.yml [hunk 1]
```diff
@@ -26,3 +26,8 @@ jobs:
             PR: ${{ github.event.pull_request.html_url }}
 
           fields: "${{ secrets.JIRA_FIELDS }}"
+      - name: Transition issue
```
Comment (MaitriGurey1) [inline .github/workflows/jira.yml:29]: What this check handles ?

File: package.json [hunk 2]
```diff
@@ -6,21 +6,21 @@
     "dist"
   ],
   "dependencies": {
-    "@contentstack/app-sdk": "^1.5.0",
```
Comment (MaitriGurey1) [inline package.json:9]: version bump is required ?

File: src/common/contexts/entrySidebarExtensionContext.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import React from "react";
 
 export type EntrySidebarExtensionContextType = {
-  entryData: { [key: string]: any };
+  entryData: { [key: string]: unknown };
```
Comment (MaitriGurey1) [inline src/common/contexts/entrySidebarExtensionContext.ts:4]: Nice use of unknown @Amitkanswal !

**Review by MaitriGurey1 — APPROVED** [review]

File: .github/workflows/jira.yml (hunk 1 above)
Comment (Amitkanswal) [reply .github/workflows/jira.yml:29]: this is sync changes not added by me

File: package.json (hunk 2 above)
Comment (Amitkanswal) [reply package.json:9]: yes

**Pull request merged**